### DISCUSSION TITLE
Problems with pst files embedded in AD1

### DISCUSSION THREAD

**Discussion by carlo-amz** [sepinf-inc/IPED#1778] [Q&A]
Hi all,
I'm experiencing some issues with pst files embedded in AD1 file and folder containers.
When I load the pst extracted form the AD1 I got no problems and IPED can parse it flawlessly.
Othwerwise if I load the AD1 file, IPED parses the other files inside the AD1 but not the PST.
Here's the warnings in the logfile

> 2023-07-15 13:18:49	[WARN]	[parsers.mail.OutlookPSTParser]			java-libpst failed, using libpff on FILENAME.ad1/EXPORTPST:E:\EXPORTPST/NOME5_Export/05.19.2023-1217PM/Exchange/accountmail1@AZIENDA.pst>>accountmail1@AZIENDA.pst_10
com.pff.PSTException: **Invalid file header: Œ”¦A, expected: !BDN**
	at com.pff.PSTFile.<init>(PSTFile.java:177) ~[java-libpst-0.9.5-20210209-p1.jar:?]
	at com.pff.PSTFile.<init>(PSTFile.java:144) ~[java-libpst-0.9.5-20210209-p1.jar:?]
	at iped.parsers.mail.OutlookPSTParser.safeParse(OutlookPSTParser.java:157) [iped-parsers-impl-4.1.2.jar:?]
	at iped.parsers.mail.OutlookPSTParser.parse(OutlookPSTParser.java:131) [iped-parsers-impl-4.1.2.jar:?]
	at org.apache.tika.parser.CompositeParser.parse(CompositeParser.java:298) [tika-core-2.4.0-p1.jar:2.4.0]
	at iped.parsers.standard.StandardParser.parse(StandardParser.java:245) [iped-parsers-impl-4.1.2.jar:?]
	at iped.engine.io.ParsingReader$BackgroundParsing.run(ParsingReader.java:247) [iped-engine-4.1.2.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]

> 2023-07-15` 13:18:51	[WARN]	[engine.data.Item]			Error closing resources of FILENAME.ad1/EXPORTPST:E:\EXPORTPST/NOME5_Export/05.19.2023-1217PM/Exchange/accountmail1@AZIENDA.pst>>accountmail1@AZIENDA.pst_10
java.nio.file.FileSystemException: G:\IPED_Default\iped-temp-8869270990689228384\iped10991152981019200119.pst: Unable to access file. The file is being used by another process.

In this case the ad1 contained different pst files and I found in the log the same problem about the invalid header for every pst.

Thank you for any help!

**Comment by lfcnassif** [comment]
> accountmail1@AZIENDA.pst_10

The _10 suffix is a tip the file was fragmented by the tool because it is very big. Please check if the parent file (without the suffix) was classified into Mailbox category and if the hash computed by the tool is correct (equal to the extracted PST hash). If not maybe the AD1 decoder has some issue for large files. Also please send the processing log.

**Comment by lfcnassif** [comment]
I'm not able to reproduce with last 4.1.3 version. I created a sample AD1 with 1 PST of 2.5GB of size and the PST was decoded fine.

Please share the processing log. I may also need a sample AD1 with a sample PST that triggers the issue, so I can reproduce to investigate. A step by step to reproduce the issue would help.

**Comment by carlo-amz** [comment]
I'm sorry but I cannot post the full log because is quite big (about 700MB) and contains sensitive data.
I tried to reproduce the error on a different PC but this time all the pst were parsed without errors.
They're about 10GB each for a total of 63 pst file inside a AD1 of more than 600GB.
Maybe the error is due the fact that I've used the append options to add this image to a case.
If I create a new case I got no problem.
The evidence is huge so it take time to perform the tests, however I'll try to reproduce the error in some way and I'll keep you posted.
Thank you for your time and support!

**Reply by lfcnassif** [reply]
You can send the log to me privatelly. Maybe the problems were due to out of space in your temp directory, IPED needs to copy PST files to temp to decode them.

**Reply by lfcnassif** [reply]
Hi @carlo-amz, sorry for my long delay. I took a look at the log you sent privately to me. There are a bunch of errors, but all (or most) of them were caused by **timeouts** while creating temp files during parsing your large PSTs. To avoid this kind of timeout errors, I would recommend:
- Use a SSD disk as temp processing disk if not used yet;
- Store your AD1 evidence in a faster disk, this is important if you have many CPU processing cores, because many PST temp files could be created at the same time and maybe the disk where the AD1 was stored wasn't able to handle many concurrent read requests, taking too long and resulting in timeouts;
- Increase the parsing timeout parameters in conf/ParsingTaskConfig.txt: `timeout` and `timeoutPerMB`

**Reply by lfcnassif** [reply]
Additionally, in above stacktrace the PST file was fragmented (leading to invalid header in non first fragments) because of the timeout. Timed out files are indexed using "strings" and large files processed by "strings" are fragmented. So timeouts are also the cause of PST fragmentation and invalid headers here.

**Comment by carlo-amz** [comment]
Hi and sorry for the late reply.Thank you so much for your assistance. I'll make treasure of your suggestions!Have a nice day!CarloInviato dal mio Galaxy
-------- Messaggio originale --------Da: Luis Filipe Nassif ***@***.***> Data: 10/08/23  03:58  (GMT+01:00) A: sepinf-inc/IPED ***@***.***> Cc: carlo-amz ***@***.***>, Mention ***@***.***> Oggetto: Re: [sepinf-inc/IPED] Problems with pst files embedded in AD1 (Discussion #1778) 
Additionally, in above stacktrace the PST file was fragmented (leading to invalid header in non first fragments) because of the timeout. Timed out files are indexed using "strings" and large files processed by "strings" are fragmented. So the timeouts are also the cause of PST fragmentation.
